### ISSUE TITLE
Unable to install on Linux Mint

### ISSUE THREAD

**Issue by Clothlen** [kayhayen/Nuitka#111]
Hi, I can't figure out this problem.

Version of Nuitka (`nuitka --version`):
```
0.5.32.7
2.7.12 (default, Dec  4 2017, 14:50:18) 
[GCC 5.4.0 20160609]
/usr/bin/python
```
`nuitka3 --version`:
```
0.5.32.7
3.7.0 (default, Sep  2 2018, 11:21:38) 
[GCC 5.4.0 20160609]
/usr/local/bin/python3.7
```
`python -V`: `Python 2.7.12`
`python3 -V` AND `python3.5 -V`: `Python 3.5.2`
`python3.6 -V`: `Python 3.6.6`
`python3.7 -V`: `Python 3.7.0`

My gcc version: `gcc -v` returns `gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.10) `

My platform is Linux Mint 18.1 Serena: `cat /etc/linuxmint/info`
```
RELEASE=18.1
CODENAME=serena
EDITION="Cinnamon 64-bit"
DESCRIPTION="Linux Mint 18.1 Serena"
DESKTOP=Gnome
TOOLKIT=GTK
NEW_FEATURES_URL=http://www.linuxmint.com/rel_serena_cinnamon_whatsnew.php
RELEASE_NOTES_URL=http://www.linuxmint.com/rel_serena_cinnamon.php
USER_GUIDE_URL=help:linuxmint
GRUB_TITLE=Linux Mint 18.1 Cinnamon 64-bit
```

I attempted to follow the instructions provided at the [downloads page](https://nuitka.net/pages/download.html#stable-packages) for Debian/Ubuntu. My `$CODENAME` was `serena` (`lsb_release -c -s`), however, which didn't appear in the [index](http://nuitka.net/deb/stable/). I figured that [since the package base of Serena was Xenial](https://linuxmint.com/download_all.php), Xenial could be used as a substitute. In addition, I also installed Nuitka via pip (`python3 -m pip install nuitka`).

I have an example program that looks like:

```
"""Runs the primary application."""
import sys
print("Hello.", sys.version)
```

I try to compile it with `nuitka3 --recurse-all --output-dir=build source/test.py && ./build/test.exe` and it worked; it compiled and was successfully run. However, it was using Python 3.5, which is not wanted. I want it to use **Python 3.7**. Curiously, after a restart, the command no longer works; I don't know why. However, `python3.5 -m nuitka --recurse-all --output-dir=build source/test.py && ./build/test.exe` seems to bring the same result as from before the restart.

I tried doing `python3.5 -m nuitka --recurse-all --python-version=3.7 --output-dir=build source/test.py && ./build/test.exe`. This brings a weird error:

```
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/nuitka/__main__.py", line 218, in <module>
    main()
  File "/usr/lib/python3/dist-packages/nuitka/__main__.py", line 60, in main
    from nuitka import Options                  # isort:skip
ModuleNotFoundError: No module named 'nuitka'
```

I tried running `python3.7 -m nuitka --recurse-all --python-version=3.7 --output-dir=build source/test.py` as well. This produced a 2960-line-long error log, so I put it on [pastebin.com](https://pastebin.com/YGpp5yR1). Running `nuitka3 --recurse-all --python-version=3.7 --output-dir=build source/test.py` produced a similar, if not identical, result. Running either with `--verbose --debug` as well gets [this smaller(?) pastebin](https://pastebin.com/kD13Dg83).

Alternatively, running `nuitka --recurse-all --python-version=3.7 --output-dir=build source/test.py` produced `RuntimeError: Bad magic number in .pyc file`.

What should I do?

**Comment by kayhayen** [comment]
The "--python-version" parameter is not what you ought to use. It can only work well if Nuitka is not installed to the system and I am going to remove it. It works under too limited circumstances.

I think for you though, the package will have installed Nuitka into a shared spot for all Python3 versions, and "python3.7 -m nuitka" is exactly how it should work.

However, your Python 3.7 fails to link against libpython from what it seems. It works on my Debian testing machine. I am pretty sure that it works if it comes come Debian alikes.

Your Python 3.7 might be a statically linked one, and thus not supported yet. See issue #95 which would deal with that. It was probably not built with "--enable-shared", which Debian normally does, it definitely is not standard part of the release, or is it?

Yours,
Kay

**Comment by Clothlen** [comment]
Thanks. I solved the problem. I installed Python 3.7 some time ago, but I compiled it myself and for whatever reason it wasn't installed to `/usr/bin/python3.7`, but some `/usr/local/bin` directory. I uninstalled it by reinstalling with checkinstall (which nearly broke Python because when it reinstalled it decided to overwrite Python2.7, and aptitude was telling me to uninstall all the dependent packages), and then uninstalling it with `dpkg`. Then I reinstalled it the correct way by adding the correct PPA and installing with aptitude instead of compiling it myself.

Running `python3.7 -m nuitka --recurse-all --output-dir=build source/test.py` now works and gives the correct output:

```
Hello. 3.7.0 (default, Aug  6 2018, 20:07:46) 
[GCC 5.4.0 20160609]
```

Thanks! I have little experience with git issues and etiquette, but I think this needs to be "closed" or "solved" now.
